internal/server/turn.go

```diff
@@ -14,37 +15,76 @@ func (s *SOMASServer) runTurn() error {
 	s.logf("TURN: %v, Season: %v", s.gameState.Turn, s.gameState.Season)
 
 	if err := s.updateIslands(); err != nil {
-		return err
+		return errors.Errorf("Error updating islands: %v", err)
 	}
 
-	if err := s.runOrgs(); err != nil {
-		return err
+	allActions := []common.Action{}
+
+	// run all orgs and get all actions
+	if actions, err := s.runOrgs(); err != nil {
+		return errors.Errorf("Error running orgs: %v", err)
+	} else {
+		allActions = append(allActions, actions...)
+	}
+
+	// get all end of turn actions
+	if actions, err := s.getEndOfTurnActions(); err != nil {
+		return errors.Errorf("Error in getEndOfTurnActions: %v", err)
+	} else {
+		allActions = append(allActions, actions...)
+	}
+
+	// !!!!!!!!!!!!
+	// TESTING ONLY
+	// !!!!!!!!!!!!
+	exampleAction := common.Action{
+		ActionType: common.GiveClientResource,
+		GiveClientResourceAction: &common.GiveClientResourceAction{
+			SourceClientID: shared.Team1,
+			TargetClientID: shared.Team2,
+			Resources:      1,
+		},
+	}
+	allActions = append(allActions, exampleAction)
+
+	// dispatch actions
+	if err := s.gameState.DispatchActions(allActions); err != nil {
+		return errors.Errorf("Error dispatching actions: %v", err)
 	}
 
 	if err := s.endOfTurn(); err != nil {

```

Comment:
yes, in here we mutate `gameState` using the server but not a function of `gameState`, but this is strictly for convenience only--actions are strictly defined as items initiated by islands + the org meetings. Server is free to mutate gameState as it wishes as long as it is a valid move.